backend.native/compiler/ir/backend.native/src/org/jetbrains/kotlin/backend/konan/lower/InteropLowering.kt

```diff
@@ -500,13 +500,13 @@ internal class InteropLoweringPart1(val context: Context) : IrBuildingTransforme
         val constructedClass = outerClasses.peek()!!
         val constructedClassDescriptor = constructedClass.descriptor
 
-        if (!constructedClassDescriptor.isObjCClass()) {
+        if (!constructedClass.isObjCClass()) {
             return expression
         }
 
-        constructedClassDescriptor.containingDeclaration.let { classContainer ->
-            if (classContainer is ClassDescriptor && classContainer.isObjCClass() &&
-                    constructedClassDescriptor == classContainer.companionObjectDescriptor) {
+        constructedClass.parent.let { parent ->
+            if (parent is IrClass && parent.isObjCClass() &&
+                    constructedClass == parent.companionObject()) {
```

Comment:
Finding companion object among `parent`'s declarations is not required here -- it is enough to check `constructedClass.isCompanion`.

Reply:
done